packages/@aws-cdk/aws-appsync/test/appsync-grant.test.ts

```diff
@@ -0,0 +1,1031 @@
+                  {
+                    Ref: 'AWS::Region',
+                  },
+                  ':',
+                  {
+                    Ref: 'AWS::AccountId',
+                  },
+                  ':apis/',
+                  {
+                    'Fn::GetAtt': [
+                      'API62EA1CFF',
+                      'ApiId',
+                    ],
+                  },
+                  '/types/Mutation/*',
+                ],
+              ],
+            },
+          },
+        ],
+      },
+    });
+  });
+
+  test('grantMutation provides correct FIELD permissions', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const role = new iam.Role(stack, 'Role', {
+      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+    });
+    const api = new appsync.GraphQLApi(stack, 'API', {
+      name: 'demo',
+      schemaDefinitionFile: join(__dirname, 'appsync.test.graphql'),
+      authorizationConfig: {
+        defaultAuthorization: {
+          authorizationType: appsync.AuthorizationType.IAM,
+        },
+      },
+    });
+
+    // WHEN
+    api.grantMutation(role, ['addTest']);
+
+    // THEN
+    expect(stack).toHaveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Action: 'appsync:graphql',
+            Effect: 'Allow',
+            Resource: {
+              'Fn::Join': [
+                '',
+                [
+                  'arn:',
+                  {
+                    Ref: 'AWS::Partition',
+                  },
+                  ':appsync:',
+                  {
+                    Ref: 'AWS::Region',
+                  },
+                  ':',
+                  {
+                    Ref: 'AWS::AccountId',
+                  },
+                  ':apis/',
+                  {
+                    'Fn::GetAtt': [
+                      'API62EA1CFF',
+                      'ApiId',
+                    ],
+                  },
+                  '/types/Mutation/fields/addTest',
+                ],
+              ],
+            },
+          },
+        ],
+      },
+    });
+  });
+});
+
+describe('grantQuery Permissions', () => {
+
+  test('grantQuery should not throw error', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const role = new iam.Role(stack, 'Role', {
+      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+    });
+    const api = new appsync.GraphQLApi(stack, 'API', {
+      name: 'demo',
+      schemaDefinitionFile: join(__dirname, 'appsync.test.graphql'),
+      authorizationConfig: {
+        defaultAuthorization: {
+          authorizationType: appsync.AuthorizationType.IAM,
+        },
+      },
+    });
+
+    // WHEN
+    const when = () => {
+      api.grantQuery(role);
+    };
+
+    // THEN
+    expect(when).not.toThrow();
+  });
+
+  test('grantQuery provides correct with NO FIELD permissions', () => {
```

Comment:
This sentence is quite confusing to me...

How about `"grantQuery provides Query/* permissions when called without the 'fields' argument"`?

Same comment for the other tests.